docs/md/executor.md

```diff
@@ -0,0 +1,163 @@
+  "methodConfigurationVersion": 1,
+  "fromSource": "importSnapshot"
+}
+```
+
+![](./assets/terra-method-configuration.png)
+
+The table below summarises the purpose of each attribute in the above request.
+
+| Attribute                    | Description                                                                                       |
+|------------------------------|---------------------------------------------------------------------------------------------------|
+| `name`                       | Selects the `Terra` executor implementation.                                                      |
+| `workspace`                  | Terra Workspace in which to execute workflows.                                                    |
+| `methodConfiguration`        | Method configuration from which to generate submissions.                                          |
+| `methodConfigurationVersion` | Expected version of the method configuration.                                                     |
+| `fromSource`                 | Instruction to coerce an output from an upstream `Source` to a type understood by this `executor`.|
+
+#### `workspace`
+A `{workspace-namespace}/{workspace-name}` string as it appears in the URL path
+in the Terra UI.
+
+Prerequisites:
+
+- The workspace must exist prior to workload creation.
+- `workflow-launcher@firecloud.org` must be a workspace "Writer" in order to
+  import snapshots to the workspace.
+- The workspace must be compatible with any downstream processing stage that
+  consumes its workflows.
+  For example, we cannot sink a workflow's outputs to an outside workspace.
+
+#### `methodConfiguration`
+A `{method-configuration-namespace}/{method-configuration-name}` string as it
+appears in the URL path in the Terra UI.
+
+Prerequisites:
+
+- The method configuration must exist within `workspace` prior to
+  workload creation.
+
+#### `methodConfigurationVersion`
+The expected version of `methodConfiguration`, stored as an integer
+in Firecloud.
+
+Prerequisites:
+
+- The `methodConfiguration` version when fetched from Firecloud should match
+  `methodConfigurationVersion`.
+  
+!!! warning "Implications of Version Mismatch"
+    A version mismatch may indicate a possible concurrent modification of the
+    method configuration used for launching submissions.  Modification is possible
+    programmatically or via the Terra UI.  An unexpected modification may cause
+    submission and/or workflow creation to fail.
+
+#### `fromSource`
+This attribute tells workflow-launcher how to coerce an output
+from an upstream `Source` into a data type understood by the executor.
+
+Prerequisites:
+
+- Must be one of the following supported coercion instructions.
+
+###### `importSnapshot`
+Workflow-launcher should import Terra Data Repository (TDR) snapshots
+into `workspace` as snapshot references,
+updating `methodConfiguration` with the reference as its root entity type.
+
+## Developer Guide
```

Comment:
I agree with @rexwangcc that I'd prefer to see the Developer Guide(s) separate.

I could see separate User Guide files for source, executor and sink since the number of implementations may grow.  But I'd be in favor of grouping the developer-facing information in a single guide.